pkg/crc/segment/segment.go

```diff
@@ -0,0 +1,79 @@
+package segment
+
+import (
+	"io/ioutil"
+	"os"
+
+	"github.com/code-ready/crc/cmd/crc/cmd/config"
+	crcConfig "github.com/code-ready/crc/pkg/crc/config"
+	"github.com/code-ready/crc/pkg/crc/constants"
+	"github.com/code-ready/crc/pkg/crc/logging"
+	"github.com/code-ready/crc/pkg/crc/version"
+	"github.com/pborman/uuid"
+	"github.com/segmentio/analytics-go"
+)
+
+type Client struct {
+	segmentClient analytics.Client
+	config        *crcConfig.Config
+}
+
+func NewClient(config *crcConfig.Config) (Client, error) {
+	client, err := analytics.NewWithConfig("cvpHsNcmGCJqVzf6YxrSnVlwFSAZaYtp", analytics.Config{
+		DefaultContext: &analytics.Context{
+			App: analytics.AppInfo{
+				Name:    constants.DefaultName,
+				Version: version.GetCRCVersion(),
+			},
+		},
+	})
+	if err != nil {
+		return Client{}, err
+	}
+	return Client{segmentClient: client, config: config}, nil
+}
+
+func (c Client) Upload(err error) error {
+	defer c.segmentClient.Close()
```

Comment:
What will happen on the second call of Upload ? segmentClient will be closed.

Reply:
Since we are hooking this with commands in case of error so every command will just going to run once the upload functionality.

Reply:
This seems like a fairly strong limitation, we should at least be documented. It would be better to provide a `Close()` method, and handle that in the caller. My understanding is that long term we may want to call this for each `crc start`, `crc stop`, ... calls, even when successful, in which case this will come and byte us. We'll also want to use this in the daemon.

Reply:
`c.segmentClient.Close()` flushes all the messages which were enqueued, I suspect this could be blocking for some time in case of DNS/network issues. One thing which might need to be improved in the future :)